linkcheck/strformat.py

```diff
@@ -56,10 +56,15 @@ def unicode_safe (s, encoding=i18n.default_encoding, errors='replace'):
     @rtype: unicode
     """
     assert s is not None, "argument to unicode_safe was None"
-    if isinstance(s, unicode):
+    if isinstance(s, str_text):
         # s is already unicode, nothing to do
         return s
-    return unicode(str(s), encoding, errors)
+
+    try:
+        return unicode(str(s), encoding, errors)
+    except NameError:  # Python3
+        if isinstance(s, bytes):
+            return s.decode("utf-8", errors)
```

Comment:
Hm, previously `unicode_safe(42)` would return `u"42"`.  With this change, on Python 3, it will return `None`.

Reply:
Added this commit:
```diff
@@ -64,7 +64,8 @@ def unicode_safe (s, encoding=i18n.default_encoding, errors='replace'):
         return unicode(str(s), encoding, errors)
     except NameError:  # Python3
         if isinstance(s, bytes):
-            return s.decode("utf-8", errors)
+            return s.decode(encoding, errors)
+        return str(s)
```
(used the encoding function argument too).